### PR TITLE
Fix cholesky

### PR DESCRIPTION
**Pull request by Rubtsowa**

### CODE REVIEW

File: dpnp/backend/kernels/dpnp_krnl_linalg.cpp [hunk 1]
```diff
@@ -34,47 +34,56 @@
 namespace mkl_blas = oneapi::mkl::blas::row_major;
 namespace mkl_lapack = oneapi::mkl::lapack;
 
-template <typename _DataType>
-class dpnp_cholesky_c_kernel;
 
-template <typename _DataType>
-void dpnp_cholesky_c(void* array1_in, void* result1, size_t* shape)
+template <typename _DataType, typename _ResultType>
```
Comment (samir-nasibli) [inline dpnp/backend/kernels/dpnp_krnl_linalg.cpp:38]: Why are you using different type names, if you instantiate the same?

File: dpnp/linalg/dpnp_iface_linalg.py [hunk 2]
```diff
@@ -86,11 +86,16 @@ def cholesky(input):
...
-        result = dpnp_cholesky(input)
+    if not use_origin_backend(input):
+        if not isinstance(input, dparray):
+            pass
+        elif input.shape[-1] != input.shape[-2]:
+            pass
+        elif input.ndim < 3:
+            pass
+        else:
+            result = dpnp_cholesky(input)
```
Comment (densmirn) [inline dpnp/linalg/dpnp_iface_linalg.py:98]: You will never fallback to numpy with this code.

File: dpnp/backend/include/dpnp_iface.hpp [hunk 3]
```diff
@@ -229,10 +229,10 @@ INP_DLLEXPORT void dpnp_sort_c(void* array, void* result, size_t size);
  *
  * @param [in]  array   Input array with data.
  * @param [out] result  Output array.
- * @param [in]  shape   Shape of input array.
+ * @param [in]  size    Number of elements in input arrays.
  */
-template <typename _DataType>
-INP_DLLEXPORT void dpnp_cholesky_c(void* array1_in, void* result1, size_t* shape);
+template <typename _DataType, typename _ResultType>
+INP_DLLEXPORT void dpnp_cholesky_c(void* array1_in, void* result1, size_t size);
```
Comment (samir-nasibli) [inline dpnp/backend/include/dpnp_iface.hpp:235]: Fix this function prototype, please.

**Comment by samir-nasibli** [comment]
Can you please, move all cython layer logic to your kernels? Just leave getting function ptr on cython level.

File: dpnp/backend/include/dpnp_iface.hpp [hunk 4]
```diff
@@ -229,10 +229,11 @@ INP_DLLEXPORT void dpnp_sort_c(void* array, void* result, size_t size);
  *
  * @param [in]  array   Input array with data.
  * @param [out] result  Output array.
- * @param [in]  shape   Shape of input array.
+ * @param [in]  size    Number of elements in input arrays.
+ * @param [in]  size_   Number of elements in each parts input arrays.
  */
 template <typename _DataType>
-INP_DLLEXPORT void dpnp_cholesky_c(void* array1_in, void* result1, size_t* shape);
+INP_DLLEXPORT void dpnp_cholesky_c(void* array1_in, void* result1, size_t size, size_t size_);
```
Comment (shssf) [inline dpnp/backend/include/dpnp_iface.hpp:236]: Please make variables names more understandable. 
These are look like a shape of input array, but not sure in it

File: dpnp/backend/kernels/dpnp_krnl_linalg.cpp [hunk 5]
```diff
@@ -34,47 +34,78 @@
 namespace mkl_blas = oneapi::mkl::blas::row_major;
 namespace mkl_lapack = oneapi::mkl::lapack;
 
-template <typename _DataType>
-class dpnp_cholesky_c_kernel;
 
 template <typename _DataType>
-void dpnp_cholesky_c(void* array1_in, void* result1, size_t* shape)
+void dpnp_cholesky_c(void* array1_in, void* result1, size_t size, size_t last_shape_elem)
```
Comment (samir-nasibli) [inline dpnp/backend/kernels/dpnp_krnl_linalg.cpp:39]: Please, declare `size` and `last_shape_elem` as `const size_t` type.

Comment (samir-nasibli) [reply dpnp/backend/kernels/dpnp_krnl_linalg.cpp:39]: And rename please `last_shape_elem`.
Some variants for you: `step_size`,  `data_size`, `block_size`, `last_dim_size`.

**Pull request merged**